src/prefect/settings.py

```diff
@@ -234,6 +234,43 @@ def warn_on_database_password_value_without_usage(values):
     return values
 
 
+def get_deprecated_prefect_cloud_url(settings, value):
+    warnings.warn(
+        "`PREFECT_CLOUD_URL` is deprecated. Use `PREFECT_CLOUD_API_URL` instead.",
+        DeprecationWarning,
+    )
+    return value or PREFECT_CLOUD_API_URL.value_from(settings)
+
+
+def check_for_deprecated_cloud_url(settings, value):
+    deprecated_value = PREFECT_CLOUD_URL.value_from(settings, bypass_callback=True)
+    if deprecated_value is not None:
+        warnings.warn(
+            "`PREFECT_CLOUD_URL` is set and will be used instead of `PREFECT_CLOUD_API_URL` for backwards compatibility. `PREFECT_CLOUD_URL` is deprecated, set `PREFECT_CLOUD_API_URL` instead.",
+            DeprecationWarning,
+        )
+    return deprecated_value or value
+
+
+def default_ui_url(settings, value):
+    if value is not None:
+        return value
+
+    # Otherwise, infer a value from the API URL
+    ui_url = api_url = PREFECT_API_URL.value_from(settings)
+
+    if not api_url:
+        return "http://ephemeral-orion"
+
+    if api_url.endswith("/api"):
+        ui_url = ui_url[:-4]
+
+    if api_url.startswith("https://api."):
+        ui_url = ui_url.replace("https://api.", "https://app.")
```

Comment:
Pure curiosity - is this a common issue?

Reply:
Hm idk about issue, this is just how we transform our API URl to a UI URL. I'm a little hesitant about having it be this broad honestly, maybe it should only replace `api.prefect.cloud` with `app.prefect.cloud`.

Reply:
Is there any downside to restricting it?

Reply:
Nope! I think we should. Otherwise we may incorrectly coerce someone's custom domain.